src/pools/abstract-pool.ts

```diff
@@ -155,6 +159,12 @@ export abstract class AbstractPool<
     this.emitter = new PoolEmitter()
   }
 
+  private checkFilePath (filePath: string) {
+    if (!filePath || filePath === '') {
```

Comment:
`!filePath` is already checking for empty string, because '' is a falsy value

```suggestion
    if (!filePath) {
```